cpp/path_module/algorithm/path.hpp

```diff
@@ -56,52 +57,102 @@ struct LabelBoolsStatus {
   bool termination_activated = false;
 };
 
-struct RelationshipSets {
-  std::unordered_set<std::string> outgoing_rel;
-  std::unordered_set<std::string> incoming_rel;
-  std::unordered_set<std::string> any_rel;
+enum RelDirection { kNone = -1, kAny = 0, kIncoming = 1, kOutgoing = 2, kBoth = 3 };
```

Comment:
Prefer scoped enums to unscoped enums, it is general practice in C++. I would suggest here to change `enum` to `enum class` and use that.